gb-dbg/src/debugger/breakpoints/breakpoint_node.rs

```diff
@@ -0,0 +1,122 @@
+use crate::dbg_interfaces::{CpuRegs, RegisterDebugOperations};
+use anyhow::anyhow;
+use nom::branch::alt;
+use nom::bytes::complete::{tag, take_while_m_n};
+use nom::combinator::map;
+use nom::sequence::tuple;
+use nom::IResult;
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+
+#[derive(Debug, Copy, Clone)]
+enum Operator {
+    Eq,
+    NotEq,
+    Sup,
+    Inf,
+    SupEq,
+    InfEq,
+}
+
+impl Display for Operator {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Operator::Eq => write!(f, "=="),
+            Operator::NotEq => write!(f, "!="),
+            Operator::Sup => write!(f, ">"),
+            Operator::Inf => write!(f, "<"),
+            Operator::SupEq => write!(f, ">="),
+            Operator::InfEq => write!(f, "<="),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct BreakpointNode {
+    lhs: CpuRegs,
+    op: Operator,
+    rhs: u16,
+}
+
+impl Display for BreakpointNode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        if let CpuRegs::PC = self.lhs {
+            if let Operator::Eq = self.op {
```

Comment:
collapse the `if` with `CpuRegs::PC == self.lhs && Operator::Eq == self.op`.
you may need to derive `Eq` & `PartialEq`